Tahi-project/tahi
# APERTA-5419: Storing metadata snapshots

Jira ticket: https://developer.plos.org/jira/browse/APERTA-5419
#### Summary

This PR introduces metadata card snapshots when a paper is submitted or re-submitted by a user.
#### How to test
- Pull down code locally.
- Prepare your database:
  - If you import **staging**'s as a starting point then you'll have the nested questions already loaded. 
  - If you do not import **staging**'s then you may want to run these rake tasks: `rake data:migrate:questions-to-nested-questions:reset_all data:migrate:questons-to-nested-questions:migrate_all`
  - If you set up a fresh DB using `rake db:setup` then the nested questions will be loaded.
- Open up a **rails console** session and have that handy. To start you should have _no snapshots_, e.g. `Snapshot.count`
- Create a paper, add all the metadata cards:
  - [x] Authors
  - [x] Competing Interests
  - [x] Data Availability
  - [x] Ethics
  - [x] Figures
  - [x] Financial Disclosure
  - [x] Publishing Related Questions
  - [x] Reporting Guidelines
  - [x] Supporting Information
  - [x] New Taxon
  - [x] Upload manuscript
- Go thru and perform all the cards, and complete them as you go so you can submit a paper.
- Finally, **submit your paper**.
- In your **rails console** session you should now have a `Snapshot` for every metadata task.
  - They should each be tied to the right task.
  - They should each be tied to the right paper.
  - They should each have a `content` field that is JSON that contains the snapshot. 
#### Reviewer tasks:
#### After the Code Review:

Author tasks:
